plugins/org.eclipse.n4js.n4mf.model/model/N4MF.xcore

```diff
@@ -25,25 +25,27 @@ package org.eclipse.n4js.n4mf
 import org.eclipse.emf.common.util.BasicEList
 import org.eclipse.n4js.utils.io.FileUtils
 
-class ProjectDescription extends SimpleProjectDescription {
+class ProjectDescription {
+	String projectId
+	String vendorId
 	String vendorName
 	contains DeclaredVersion projectVersion
 	ProjectType projectType
 	String mainModule
-	contains ExtendedRuntimeEnvironment extendedRuntimeEnvironment
-	contains ProvidedRuntimeLibraries providedRuntimeLibraries
-	contains RequiredRuntimeLibraries requiredRuntimeLibraries
-	contains ProjectDependencies projectDependencies
+	contains ProjectReference extendedRuntimeEnvironment
+	contains ProjectReference[] providedRuntimeLibraries
+	contains ProjectReference[] requiredRuntimeLibraries
+	contains ProjectDependency[] projectDependencies
 	String implementationId
-	contains ImplementedProjects implementedProjects
-	contains InitModules initModules
-	contains ExecModule execModule
+	contains ProjectReference[] implementedProjects
+	contains BootstrapModule[] initModules
+	contains BootstrapModule execModule
 	String outputPathRaw
 	String[] libraryPathsRaw
 	String[] resourcePathsRaw
-	contains SourceFragment[] sourceFragment
+	contains SourceContainerDescription[] sourceContainers
 	contains ModuleFilter[] moduleFilters
-	contains TestedProjects testedProjects
+	contains ProjectDependency[] testedProjects
```

Comment:
Why are tested projects of type `ProjectDependency`? I think, it should be:

`contains ProjectReference[] testedProjects`

Reply:
Depends. Different versions of `TestProject` can be testing different versions of `TestedProject`
eg. `TestedProject, v=1.2.3` might be testing `TestedProject, v=4.5.6`

Unless we remove this concept, it has to be `ProjectDependency`

Reply:
good point Quang, I ran into the exact same question today while further working on this;
I have notes of discussing this further, for now it will stay `ProjectDependency` (which it was before; we just moved the property, did not change its type)

In general, I agree with Jakub (i.e. we need the version there).